compiler/src/iree/compiler/InputConversion/StableHLO/Preprocessing/Canonicalization.cpp

```diff
@@ -1048,6 +1048,42 @@ struct ZeroExtentTensorCanon final : RewritePattern {
   }
 };
 
+template <typename ElementwiseOpT>
+struct ReorderElementwiseAndShapeOp final : OpRewritePattern<ElementwiseOpT> {
+  using OpRewritePattern<ElementwiseOpT>::OpRewritePattern;
+  LogicalResult matchAndRewrite(ElementwiseOpT op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    auto definingOp = op.getOperand().getDefiningOp();
+    if (!definingOp) {
+      return failure();
```

Comment:
Could you include a `notifyMatchFailure` for the failure cases?

Reply:
Done